### PR TITLE
Refactor logic to a separate Github actions (public and private)

### LINKED ISSUE #2393: Remove Docker image upload to Amazon ECR on backups
- [x] Delete  "Deploy to Amazon ECR" Github action (`.github/workflows/build_and_push.yml`)
- [x] Remove all GS upload references. We are not using them.
- [x] Move filtered and full backups to independent jobs

### PR DESCRIPTION
**Pull request by pdcp1**
This closes https://github.com/responsible-ai-collaborative/aiid/issues/2393

### CODE REVIEW

**Comment by pdcp1** [comment]
@kepae For some reason I can't assign this to you for review. Please review and merge it if you want to test.

File: .github/workflows/run_private_backup.yml [hunk 1]
```diff
@@ -17,41 +15,50 @@ on:
...
 
-      - name: Generate public and private backups
-        id: build-image
+      - name: Generate private backup
         run: |
-          ls
-          pwd
-          ls -a
           docker run --rm \
+            -e IS_PUBLIC_BACKUP="true" \
```
Comment (kepae) [inline .github/workflows/run_private_backup.yml]: false?

**Review by kepae — CHANGES_REQUESTED** [review]
Much cleaner to maintain, IMHO!

File: .github/workflows/run_private_backup.yml (hunk 1 above)
Comment (pdcp1) [reply .github/workflows/run_private_backup.yml]: You're right. Good catch!

File: bin/backup.sh [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  mongodump -o ${TARGET} --uri=${MONGODB_URI} 
+
+  echo "Dump MongoDB 'translations' database..."
+  mongodump -o ${TARGET} --uri=${MONGODB_URI_TRANSLATIONS}
+fi
+
+echo "Report contents are subject to their own intellectual property rights. Unless otherwise noted, the database is shared under (CC BY-SA 4.0). See: https://creativecommons.org/licenses/by-sa/4.0/" > ${TARGET}/license.txt
+
+# run tar command
+echo "Start backup ${TARGET} into ${TARGET_BUCKET_URL} ..."
```
Comment (kepae) [inline bin/backup.sh:112]: can you review how `TARGET_BUCKET_URL` is configured in our workflow? It seems that this is set to `vars.AWS_S3_{PRIVATE|PUBLIC}_BUCKET`, so its usage here is misleading if we are copying to R2.

(Please let me know if I'm missing something among the scripts/config!)

Comment (pdcp1) [reply bin/backup.sh:112]: I kept the `TARGET_BUCKET_URL` name for the AWS S3 backups. Maybe is a bit confusing.
For R2 we use different variables, all of them with prefix `CLOUDFLARE_`. Cloudflare doesn't use a URL for backups.
The variables that we should set are:
- `CLOUDFLARE_ACCOUNT_ID`
- `CLOUDFLARE_R2_ACCESS_KEY`
- `CLOUDFLARE_R2_SECRET_KEY`
- `CLOUDFLARE_R2_PUBLIC_BUCKET`
- `CLOUDFLARE_R2_PRIVATE_BUCKET`

Comment (kepae) [reply bin/backup.sh:112]: Thanks for the explanation! All those values are set, so let's see if the behavior works as expected.

**Pull request merged**